lizmap/modules/admin_api/controllers/project_rest.classic.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * @author    3liz.com
+ * @copyright 2011-2025 3Liz
+ *
+ * @see      https://3liz.com
+ *
+ * @license   https://www.mozilla.org/MPL/ Mozilla Public Licence
+ */
+
+use LizmapApi\Credentials;
+use LizmapApi\Error;
+use LizmapApi\RestApiCtrl;
+
+class project_restCtrl extends RestApiCtrl
+{
+    /**
+     * Retrieves project information based on the provided parameters.
+     * If a specific project is requested, detailed information are returned.
+     * Otherwise, a list of available project and their basic information is returned.
+     *
+     * @return object a JSON response object containing project data
+     */
+    public function get(): object
+    {
+        /** @var jResponseJson $rep */
+        $rep = $this->getResponse('json');
+
+        if (!Credentials::handle()) {
+            return Error::setError($rep, 401);
+        }
+
+        try {
+            $repo = lizmap::getRepository($this->param('repo'));
+
+            if ($repo == null) {
+                throw new Exception("Valid repository is needed", 404);
```

Comment:
Ok for me, 

We can also use `jHttp404NotFoundException`  : https://github.com/jelix/jelix/blob/959d8a9c721fd2f8720ec91102fefeb287e79ae9/lib/jelix/core/jHttpErrorException.class.php#L100

Reply:
This change needs to be done in the master branch, in the original PR #5486 
This PR is for 3.8.